lib/vkloader.c

```diff
@@ -1763,18 +1858,53 @@ generateMipmaps(ktxVulkanTexture* vkTexture, ktxVulkanDeviceInfo* vdi,
  * @brief Destructor for the object returned when loading a texture image.
  *
  * Frees the Vulkan resources created when the texture image was loaded.
+ * If a complete set of (hopefully thread-safe) suballocator callbacks
+ * are provided, it will use the free call from those instead.
+ *
+ * @param vkTexture                pointer to the ktxVulkanTexture to be destructed.
+ * @param device                   handle to the Vulkan logical device to which the texture was
+ *                                 loaded.
+ * @param pAllocator               pointer to the allocator used during loading.
+ * @param subAllocatorCallbacks    potential function pointer for guarded suballocator callbacks.
+ *                                 Pass NULL if a suballocator was not used.
+ */
+void
+ktxVulkanTexture_Destruct_WithPotentialSuballocator(ktxVulkanTexture* vkTexture, VkDevice device,
+                                                    const VkAllocationCallbacks* pAllocator,
+                                                    ktxVulkanTexture_subAllocatorCallbacks* subAllocatorCallbacks)
+{
+    ktx_bool_t useSuballocator = (subAllocatorCallbacks &&
+                                  subAllocatorCallbacks->allocMemFuncPtr &&
+                                  subAllocatorCallbacks->bindBufferFuncPtr &&
+                                  subAllocatorCallbacks->bindImageFuncPtr &&
+                                  subAllocatorCallbacks->memoryMapFuncPtr &&
+                                  subAllocatorCallbacks->memoryUnmapFuncPtr &&
+                                  subAllocatorCallbacks->freeMemFuncPtr);
+
+    vkTexture->vkDestroyImage(device, vkTexture->image, pAllocator);
+    if (useSuballocator)
```

Comment:
Same comment I raised at line 836.

Reply:
👍

Reply:
Actually regarding this... what should I do here since `Destruct` has no return value. Should the inner call return a `ktx_error_code_e` while the outer one returns void?

Reply:
I think this is fine but, as in line with another comment I just made w.r.t. VkUploadEx_WithSubAllocator use KTX_INVALID_VALUE.

Reply:
Just changed both the specs and the return value.